tez-dag/src/test/java/org/apache/tez/dag/app/dag/impl/TestTaskImpl.java

```diff
@@ -982,6 +982,23 @@ public void testSpeculatedThenRetroactiveFailure() {
     Assert.assertEquals(mockDestId, newAttempt.getSchedulingCausalTA());
   }
 
+  @Test(timeout = 20000)
+  public void testIgnoreSpeculation() {
+    TezTaskID taskId = getNewTaskID();
+    scheduleTaskAttempt(taskId);
+    MockTaskAttemptImpl firstAttempt = mockTask.getLastAttempt();
+    launchTaskAttempt(firstAttempt.getID());
+    // Have the first task succeed
+    updateAttemptState(firstAttempt, TaskAttemptState.SUCCEEDED);
+    firstAttempt.handle(new TaskAttemptEvent(firstAttempt.getID(), TaskAttemptEventType.TA_DONE));
+
+    // Ignore speculation
```

Comment:
// verify the speculation scheduding is ignored and no speculative attempt was added to the task

Reply:
done